### PR TITLE
Implementation Azure Kinect Image Provider

### LINKED ISSUE #8: AzureKinect Skeleton CSV VideoImageProvider
- [x] create VideoImageProvider for AzureKinect that uses Skeleton data from a CSV
- [x] tests
- [x] skeleton drawing

### PR DESCRIPTION
**Pull request by justamad**
Closes #8

### CODE REVIEW

File: sevivi/config/config_types/sensor_config.py [hunk 1]
```diff
@@ -57,6 +57,8 @@ class JointSynchronizedSensorConfig(SensorConfig):
     """Source columns from the camera skeleton to synchronize to"""
     sensor_sync_column_selection: Union[str, List[str]] = None
     """Columns from the sensor data to synchronize to"""
+    sync_joint_name: str = None
+    """Name of the sync sensor"""
```
Comment (enra64) [inline sevivi/config/config_types/sensor_config.py:61]: Isn't this the name of the joint from the camera we should sync to?

File: sevivi/image_provider/video_provider/azure_image_provider.py [hunk 2]
```diff
@@ -1,39 +1,141 @@
...
+    ):
         if not os.path.exists(video_path):
             raise Exception(f"File {video_path} not found!")
 
-        self.__video_capture = cv2.VideoCapture(video_path)
+        self.__video = cv2.VideoCapture(video_path)
+        self.__joint_df_3d = self._drop_duplicate_body_indices_and_confidence_values(
+            pd.read_csv(joint_3d_df, sep=";", index_col=0)
+        )
+        self.__joint_df_3d.index = pd.to_datetime(
```
Comment (enra64) [inline sevivi/image_provider/video_provider/azure_image_provider.py:35]: why not just do `pd.to_datetime(self.__joint_df_3d.index, unit="us")`?

File: sevivi/image_provider/video_provider/azure_image_provider.py [hunk 3]
```diff
@@ -1,39 +1,141 @@
...
+
 class AzureProvider(VideoImageProvider):
-    def __init__(self, video_path: str, data: str):
+    """The Azure Provider provides images from a video or path with images along"""
+
+    def __init__(
+        self,
+        video_path: str,
+        joint_3d_df: str,
+        joint_2d_df: str = None,
```
Comment (enra64) [inline sevivi/image_provider/video_provider/azure_image_provider.py:26]: Can be typed as `Optional[str]`

File: sevivi/image_provider/video_provider/azure_image_provider.py [hunk 4]
```diff
@@ -1,39 +1,141 @@
...
+                return self.__joint_df_3d[column_names]
+
+        if type(column_names) is list:
+            if not all(elem in self.__joint_df_3d.columns for elem in column_names):
+                raise KeyError(f"Missing sync joints in Dataframe: {column_names}.")
+            return self.__joint_df_3d[
+                [col for col in column_names if col in self.__joint_df_3d.columns]
+            ]
+
+        raise TypeError(f"Wrong type given: {type(column_names)}, expected: List[str]")
```
Comment (enra64) [inline sevivi/image_provider/video_provider/azure_image_provider.py:132]: expected: List[str] or str

File: sevivi/config/config_types/sensor_config.py (hunk 1 above)
Comment (justamad) [reply sevivi/config/config_types/sensor_config.py:61]: I thought so, but I guess it's redundant with the property 'camera_joint_sync_column_selection' in the config file. By using this, we can synchronize to a specific 1D axis only.

Comment (enra64) [reply sevivi/config/config_types/sensor_config.py:61]: I think we need to have a talk about this, I don't get what you're saying :D

**Pull request merged**